python/src/cucumber_messages/_messages.py

```diff
@@ -0,0 +1,954 @@
+     inclusive.
+    """
+    nanos: int
+    seconds: int
+
+
+@dataclass
+class Envelope:
+    """
+    When removing a field, replace it with reserved, rather than deleting the line.
+     When adding a field, add it to the end and increment the number by one.
+     See https://developers.google.com/protocol-buffers/docs/proto#updating for details
+
+    *
+     All the messages that are passed between different components/processes are Envelope
+     messages.
+    """
+    attachment: Optional["Attachment"] = None
+    gherkin_document: Optional["GherkinDocument"] = None
+    hook: Optional["Hook"] = None
+    meta: Optional["Meta"] = None
+    parameter_type: Optional["ParameterType"] = None
+    parse_error: Optional["ParseError"] = None
+    pickle: Optional["Pickle"] = None
+    source: Optional["Source"] = None
+    step_definition: Optional["StepDefinition"] = None
+    test_case: Optional["TestCase"] = None
+    test_case_finished: Optional["TestCaseFinished"] = None
+    test_case_started: Optional["TestCaseStarted"] = None
+    test_run_finished: Optional["TestRunFinished"] = None
+    test_run_hook_finished: Optional["TestRunHookFinished"] = None
+    test_run_hook_started: Optional["TestRunHookStarted"] = None
+    test_run_started: Optional["TestRunStarted"] = None
+    test_step_finished: Optional["TestStepFinished"] = None
+    test_step_started: Optional["TestStepStarted"] = None
+    undefined_parameter_type: Optional["UndefinedParameterType"] = None
+
+
+@dataclass
+class Exception:
+    """
+    A simplified representation of an exception
+    """
+    """
+    The type of the exception that caused this result. E.g. "Error" or "org.opentest4j.AssertionFailedError"
+    """
+    type: str
+    """
+    The message of exception that caused this result. E.g. expected: "a" but was: "b"
+    """
+    message: Optional[str] = None
+    """
+    The stringified stack trace of the exception that caused this result
+    """
+    stack_trace: Optional[str] = None
+
+
+@dataclass
+class GherkinDocument:
+    """
+    *
+     The [AST](https://en.wikipedia.org/wiki/Abstract_syntax_tree) of a Gherkin document.
+     Cucumber implementations should *not* depend on `GherkinDocument` or any of its
+     children for execution - use [Pickle](#io.cucumber.messages.Pickle) instead.
+
+     The only consumers of `GherkinDocument` should only be formatters that produce
+     "rich" output, resembling the original Gherkin document.
+    """
+    """
+    All the comments in the Gherkin document
+    """
+    comments: list["Comment"]
+    feature: Optional["Feature"] = None
+    """
+    *
+     The [URI](https://en.wikipedia.org/wiki/Uniform_Resource_Identifier)
+     of the source, typically a file path relative to the root directory
+    """
+    uri: Optional[str] = None
+
+
+@dataclass
+class Background:
+    description: str
+    id: str
+    keyword: str
+    """
+    The location of the `Background` keyword
+    """
```

Comment:
[nit] can we inline this (and all the other similar ones?

Reply:
actually, we should use comments, see https://github.com/cucumber/messages/pull/165/files#r1912240401